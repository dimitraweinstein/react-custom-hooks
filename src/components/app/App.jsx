import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import HeyArnoldDetails from '../../containers/HeyArnoldDetailList';
import HeyArnoldList from '../../containers/HeyArnoldList';
import Header from '../Header';


export default function App() {
  return (
    <BrowserRouter>
      <Header />
      <Switch>
        <Route path="/:id">
          <HeyArnoldDetails />
        </Route>
        <Route path="/">
          <HeyArnoldList />
        </Route>
      </Switch>
    </BrowserRouter>
  );
}
