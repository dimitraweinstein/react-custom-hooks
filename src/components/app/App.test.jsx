import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('renders app', async () => {
    render(<App />);

    screen.getByAltText('loading spinner');

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).not.toBeEmptyDOMElement();

  });
});

