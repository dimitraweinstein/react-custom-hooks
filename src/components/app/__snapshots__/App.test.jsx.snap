// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        aria-label="characters"
      >
        <li>
          <figure>
            <img
              alt="Nancy"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
            />
            <figcaption>
              Nancy
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Grandpa's grandfather"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
            />
            <figcaption>
              Grandpa's grandfather
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Grandpa Phil"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
            />
            <figcaption>
              Grandpa Phil
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Grandpa's father"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
            />
            <figcaption>
              Grandpa's father
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Phoebe Heyerdahl"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
            />
            <figcaption>
              Phoebe Heyerdahl
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Eugene Horowitz"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
            />
            <figcaption>
              Eugene Horowitz
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Ray"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
            />
            <figcaption>
              Ray
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Shelley"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
            />
            <figcaption>
              Shelley
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Lila Sawyer"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
            />
            <figcaption>
              Lila Sawyer
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Uncle Stinky"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
            />
            <figcaption>
              Uncle Stinky
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Arnie"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
            />
            <figcaption>
              Arnie
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Olga Pataki"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
            />
            <figcaption>
              Olga Pataki
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Timberly Johanssen"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
            />
            <figcaption>
              Timberly Johanssen
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Jerry Berman"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
            />
            <figcaption>
              Jerry Berman
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Rhonda Wellington Lloyd"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
            />
            <figcaption>
              Rhonda Wellington Lloyd
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Stinky Peterson"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
            />
            <figcaption>
              Stinky Peterson
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Kyo Heyerdahl"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
            />
            <figcaption>
              Kyo Heyerdahl
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Iggy"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
            />
            <figcaption>
              Iggy
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Mr. Sawyer"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
            />
            <figcaption>
              Mr. Sawyer
            </figcaption>
          </figure>
        </li>
        <li>
          <figure>
            <img
              alt="Thaddeus \\"Curly\\" Gammelthorpe"
              src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
            />
            <figcaption>
              Thaddeus "Curly" Gammelthorpe
            </figcaption>
          </figure>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      aria-label="characters"
    >
      <li>
        <figure>
          <img
            alt="Nancy"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
          />
          <figcaption>
            Nancy
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Grandpa's grandfather"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
          />
          <figcaption>
            Grandpa's grandfather
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Grandpa Phil"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
          />
          <figcaption>
            Grandpa Phil
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Grandpa's father"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
          />
          <figcaption>
            Grandpa's father
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Phoebe Heyerdahl"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
          />
          <figcaption>
            Phoebe Heyerdahl
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Eugene Horowitz"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
          />
          <figcaption>
            Eugene Horowitz
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Ray"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
          />
          <figcaption>
            Ray
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Shelley"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
          />
          <figcaption>
            Shelley
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Lila Sawyer"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
          />
          <figcaption>
            Lila Sawyer
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Uncle Stinky"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
          />
          <figcaption>
            Uncle Stinky
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Arnie"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
          />
          <figcaption>
            Arnie
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Olga Pataki"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
          />
          <figcaption>
            Olga Pataki
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Timberly Johanssen"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
          />
          <figcaption>
            Timberly Johanssen
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Jerry Berman"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
          />
          <figcaption>
            Jerry Berman
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Rhonda Wellington Lloyd"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
          />
          <figcaption>
            Rhonda Wellington Lloyd
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Stinky Peterson"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
          />
          <figcaption>
            Stinky Peterson
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Kyo Heyerdahl"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
          />
          <figcaption>
            Kyo Heyerdahl
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Iggy"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
          />
          <figcaption>
            Iggy
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Mr. Sawyer"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
          />
          <figcaption>
            Mr. Sawyer
          </figcaption>
        </figure>
      </li>
      <li>
        <figure>
          <img
            alt="Thaddeus \\"Curly\\" Gammelthorpe"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
          />
          <figcaption>
            Thaddeus "Curly" Gammelthorpe
          </figcaption>
        </figure>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
